// run-deepseek-cli/dist/api-local.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocalDeepSeekAPI = void 0;
// src/api-local.ts
const axios_1 = __importDefault(require("axios"));
class LocalDeepSeekAPI {
    baseUrl = 'http://localhost:11434/api';
    async complete(prompt) {
        try {
            const response = await axios_1.default.post(`${this.baseUrl}/generate`, {
                model: 'deepseek-coder:6.7b',
                prompt: prompt,
                stream: false
            });
            return response.data.response;
        }
        catch (error) {
            throw new Error('Local model error. Ensure Ollama is running.');
        }
    }
    async chat(messages) {
        try {
            const response = await axios_1.default.post(`${this.baseUrl}/chat`, {
                model: 'deepseek-coder:6.7b',
                messages: messages,
                stream: false
            });
            return response.data.message.content;
        }
        catch (error) {
            throw new Error('Local model error. Ensure Ollama is running.');
        }
    }
}
exports.LocalDeepSeekAPI = LocalDeepSeekAPI;
