// run-deepseek-cli/dist/api.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeepSeekAPI = void 0;
const axios_1 = __importDefault(require("axios"));
class DeepSeekAPI {
    config;
    constructor(config) {
        this.config = config;
    }
    async complete(prompt) {
        try {
            const response = await axios_1.default.post(this.config.apiUrl, {
                model: this.config.model,
                messages: [
                    {
                        role: 'system',
                        content: 'You are DeepSeek Coder, an AI programming assistant. Help with coding tasks, provide clear code examples, and follow best practices.'
                    },
                    {
                        role: 'user',
                        content: prompt
                    }
                ],
                temperature: 0.1,
                max_tokens: 4096
            }, {
                headers: {
                    'Authorization': `Bearer ${this.config.apiKey}`,
                    'Content-Type': 'application/json'
                },
                timeout: 30000 // 30 segundos
            });
            return response.data.choices[0].message.content;
        }
        catch (error) {
            if (error.response?.status === 401) {
                throw new Error('Invalid API key. Please check your DEEPSEEK_API_KEY.');
            }
            if (error.response?.status === 429) {
                throw new Error('Rate limit exceeded. Please try again later.');
            }
            if (error.response?.status === 400) {
                console.error('API Error Details:', error.response?.data);
                throw new Error(`Bad request: ${error.response?.data?.error?.message || 'Invalid request format'}`);
            }
            throw new Error(`API error: ${error.message}`);
        }
    }
}
exports.DeepSeekAPI = DeepSeekAPI;
